File: Perfect-Number/PN.py
from math import sqrt


class Solution(object):

    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        result = 0
        if num < 0:
            return False
        elif num == 0:
            return False
        elif num == 1:
            return False

        for i in range(1, int(sqrt(num)) + 1):
            if num % i == 0:
                print(i, num / i)
                result += i + num / i
        if result - num == num:
            return True
        else:
            return False

File: Perfect-Number/PN.rs
fn checkPerfectNumber(num: u32) -> bool {
    if num <= 1 {
        return false;
    }

    let mut result = 0;
    let x = num as f32;
    for i in 1..1 + x.sqrt() as u32 {
        if num % i == 0 {
            println!("{}", num / i);
            result += i + num / i;
        }
        //println!("{}", i);
    }
    if result - num == num {
        return true;
    }
    //println!("{}", x.sqrt() as u32);
    return false;
}

fn main() {
    println!("{}", checkPerfectNumber(10));
    println!("{}", checkPerfectNumber(28));
}
